qpylib/abstract_qpylib.py

```diff
@@ -120,48 +70,88 @@ def map_log_level(self, log_level='INFO'):
             'CRITICAL': logging.CRITICAL,
         }.get(log_level, logging.INFO)
 
+    # ==== App details ====
+
     @abstractmethod
-    def add_log_handler(self, loc_logger):
+    def get_app_id(self):
         pass
 
-    def create_log(self):
-        global logger
-        global LOGGER_NAME
-        logger = logging.getLogger(LOGGER_NAME)
-        self.add_log_handler(logger)
-        self.log("Created log " + LOGGER_NAME, 'info')
+    @abstractmethod
+    def get_app_name(self):
+        pass
 
-    def set_log_level(self, log_level='INFO'):
-        logger.setLevel(self.map_log_level(log_level))
+    def get_manifest_json(self):
+        if self.cached_manifest is None:
+            full_manifest_location = self.get_root_path(self._get_manifest_location())
+            with open(full_manifest_location) as manifest_file:
+                self.cached_manifest = json.load(manifest_file)
+        return self.cached_manifest
 
-    def log(self, message, level='info'):
-        log_fn = self.choose_log_level(level)
-        log_fn("127.0.0.1 " +
-               "[APP_ID/" +  self.get_app_id() + "]" +
-               "[NOT:" +  self.map_notification_code(level) + "] " +
-               message)
+    @abstractmethod
+    def _get_manifest_location(self):
+        pass
+
+    def _get_manifest_field_value(self, key, default_value=None):
+        manifest = self.get_manifest_json()
+        if key in manifest.keys():
+            return manifest[key]
+        if default_value is not None:
+            return default_value
+        raise KeyError('{0} is a required manifest field'.format(key))
+
+    def get_store_path(self, relative_path):
+        if relative_path == '':
+            return os.path.join(self._root_path(), 'store')
+        return os.path.join(self._root_path(), 'store', relative_path)
+
+    def get_root_path(self, relative_path):
+        return os.path.join(self._root_path(), relative_path)
 
     @abstractmethod
-    def get_console_address(self):
+    def _root_path(self):
```

Comment:
root_path was only ever there for SDK support, do we still need it?

Reply:
Probably not. Next step is to move/remove SDK-specific stuff.